ggshield/verticals/secret/extended_match.py

```diff
@@ -58,16 +69,46 @@ def from_match(
         cls, match: Match, lines: List[Line], is_patch: bool
     ) -> "ExtendedMatch":
         span = MatchSpan.from_match(match, lines, is_patch)
+
         start_line = lines[span.line_index_start]
         end_line = lines[span.line_index_end]
         line_index_start = start_line.pre_index or start_line.post_index
         line_index_end = end_line.pre_index or end_line.post_index
         assert line_index_start is not None and line_index_end is not None
         line_index_start += int(is_patch) - 1  # convert to 0-based
         line_index_end += int(is_patch) - 1
+
+        lines_with_secret = lines[span.line_index_start : span.line_index_end + 1]
+
+        match_split_lines = match.match.splitlines()
+        assert len(match_split_lines) == len(lines_with_secret)
+        if is_patch and len(lines_with_secret) > 1:
+            # The secret is multi-line and in patch, we can have the first character to remove
+            stripped_match = "\n".join(
+                (
+                    str_line_match[1:]
+                    if index > 0 and line.category != LineCategory.DATA
```

Comment:
I thought about it more and we don't need to check because the range just intersect so if `len(str_line_match)==0` then it will just return the empty line which is ok !